trikot-streams/streams/src/commonTest/kotlin/com/mirego/trikot/streams/reactive/processors/ScanWithProcessorTests.kt

```diff
@@ -122,6 +169,25 @@ class CollectProcessorTests {
         assertEquals(expectedException, receivedException)
     }
 
+    @Test
+    fun testThrowingSupplier() {
+        val publisher = Publishers.behaviorSubject("a")
+
+        @Suppress("ASSIGNED_BUT_NEVER_ACCESSED_VARIABLE")
+        var receivedException: StreamsProcessorException? = null
+
+        val scanPublisher = publisher.scanWith({ throw IllegalStateException() }) { acc, _ -> acc }
+
+        assertFailsWith(IllegalStateException::class) {
+            scanPublisher.subscribe(
+                CancellableManager(),
+                onNext = { },
+                onError = { receivedException = it as StreamsProcessorException }
```

Comment:
You should assert instead

Maybe you could use one of the 2 following functions?
```

inline fun <T : Any, reified R : T> assertIsInstance(actual: T?, message: String? = null): R {
    assertTrue(actual is R, message)
    return actual
}

inline fun <reified R> Any.assertThisIsInstance(message: String? = null): R {
    assertTrue(this is R, message)
    return this
}
```